libs/design/src/atoms/form/form-field/form-field/form-field.component.spec.ts

```diff
@@ -46,15 +62,80 @@ describe('DaffFormFieldComponent', () => {
     expect(hostElement.classList.contains('daff-form-field')).toBeTruthy();
   });
 
-  it('should render [daff-input]', () => {
-    const daffInputElement = fixture.debugElement.query(By.css('[daff-input]'));
+  describe('when the child control is in an error state', () => {
+    it('should set the `daff-error` class on the `daff-form-field__control`', () => {
+      wrapper.formControl.markAsTouched();
+      wrapper.formControl.setValue("");
+      fixture.detectChanges();
+
+      expect(wrapper.formControl.errors).toBeTruthy();
+      expect(formFieldControlElement.classList.contains("daff-error")).toEqual(true);
+    });
+  });
+
+  describe('when the control is not an in error state', () => {
+    it('should NOT set the `daff-error` class on the `daff-form-field__control`', () => {
+      wrapper.formControl.markAsTouched();
+      wrapper.formControl.setValue("Something Valid");
+      fixture.detectChanges();
+
+      expect(wrapper.formControl.errors).toBeFalsy();
+      expect(formFieldControlElement.classList.contains("daff-error")).toEqual(false);
+    });
+  });
+
+  describe('when the child control is in a valid state', () => {
+    it('should set the `daff-valid` class on the `daff-form-field__control`', () => {
+      wrapper.formControl.markAsTouched();
+      wrapper.formControl.setValue("Something Valid");
+      fixture.detectChanges();
 
-    expect(daffInputElement).toBeDefined();
+      expect(wrapper.formControl.valid).toBeTruthy();
+      expect(formFieldControlElement.classList.contains("daff-valid")).toEqual(true);
+    });
   });
 
-  it('should not render random content', () => {
-    const randomContentElement = fixture.debugElement.query(By.css('.random_content'));
+  describe('when the control is not in a valid state', () => {
+    it('should NOT set the `daff-valid` class on the `daff-form-field__control`', () => {
+      wrapper.formControl.markAsTouched();
+      wrapper.formControl.setValue("");
+      fixture.detectChanges();
 
-    expect(randomContentElement).toBeNull();
+      expect(wrapper.formControl.valid).toBeFalsy();
+      expect(formFieldControlElement.classList.contains("daff-valid")).toEqual(false);
+    });
   });
+
```

Comment:
I think tests should be added for the submit cases. Like when a form is submitted, but there's no content in the field. Right now, I think submitting a blank form wouldn't show errors on the input boxes

Reply:
@lderrickable I disagree. As the [formSubmitted] property is now deprecated on the `DaffInput` and `DaffNativeSelect` (they don't even do anything at the moment) I think it makes more sense to write docs explaining the deprecation, and how we plan to use `markAllAsTouched`. It is definitely breaking in a sense, but I wouldn't want to promote a "bad" API.

I left the `[formSubmitted]` in there so as to not cause any additional breaking changes (the PR was already becoming quite large).

Reply:
I actually didn't think that submitting the form would result in an error, but it does.